### ISSUE TITLE
More flexible definition of units of measurement

### ISSUE THREAD

**Issue by andrewlee94** [IDAES/idaes-pse#682]
Derives from #650 

So, to summarize the discussion:

1. To allow custom units in models, we would need to add convert statements to basically every term in every constraint in the core IDAES models. Whilst individually not hard, this is a large volume of work for which we currently do not have the resources.

2. Additionally, adding all these convert statements adds a very significant computational overhead in Pyomo at the moment. We have a propose solution to cut this down by adding `unit_hints` to named Expressions which we would trust for unit conversion and only check during consistency assertion, but this feature needs to be implemented in Pyomo, and then we would need to add these `unit_hints` in the IDAES code.

So, in short, we have a path forward, but we lack the resources to implement it right now, and it is probably not a short path.
